core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/DatasetResource.scala

```diff
@@ -1021,4 +1022,62 @@ class DatasetResource {
       Response.ok(streamingOutput).`type`(contentType).build()
     })
   }
+
+  @GET
+  @Path("/version-zip")
+  def retrieveDatasetVersionZip(
+      @QueryParam("path") pathStr: String,
+      @Auth user: SessionUser
+  ): Response = {
+    val uid = user.getUid
+    val decodedPathStr = URLDecoder.decode(pathStr, StandardCharsets.UTF_8.name())
+    val (_, dataset, dsVersion, _) = resolveFilePath(Paths.get(decodedPathStr))
```

Comment:
this function will throw error if the file path doesn't match the certain format. And it requires the path to be a filepath. However in this function, only `/ownerEmail/datasetName/versionName` is required. No file path should be included. So I suggest you add a bool parameter, `shouldContainFile`, to this `resolveFilePath` function(possibly rename it to `resolvePath` to make it more general-purpose).